addons/website_blog/static/src/js/systray_items/edit.js

```diff
@@ -0,0 +1,40 @@
+/** @odoo-module **/
+
+import { EditSystray } from '@website/systray_items/edit';
+import { patch } from 'web.utils';
+import wUtils from 'website.utils';
+
+patch(EditSystray.prototype, 'website_blog_edit', {
+    /**
+     * @override
+     */
+    getElements() {
+        const elements = this._super();
+        const { metadata: { mainObject } } = this.websiteService.currentWebsite;
+        if (mainObject.model === 'blog.post') {
+            return [...elements, {
+                title: this.env._t('Duplicate'),
+                callback: () => this.duplicate(),
+                icon: 'fa-clone',
+            }];
+        }
+        return elements;
+    },
+    async duplicate() {
+        if (this.websiteService.currentWebsite.metadata.mainObject.model === 'blog.post') {
+            const { metadata: { mainObject } } = this.websiteService.currentWebsite;
+            wUtils.sendRequest('/blog/post_duplicate', { blog_post_id: mainObject.id });
+        } else {
+            this._super();
+        }
+    },
```

Comment:
Red flag, this whole part is duplicate code in hr job
And it will be duplicated everytime we want to make a model able to be duplicated.

Move this in the `EditSystray` and make it used only by the hr job and blog, so it will be just 2-3 lines in case a model want to add the duplicate capability

Also, as the code will be generic in editsystay, make it generic in python too.
Eg, create a controller in website `/website/duplicate` like we have for `/website/publish`, I'll let you check.
You could then move the logic about finding a new unique name inside it (even if there is no uniq constraint on the model, it will still be nice to not have the exact same name on duplicate) + redirect to the website_url if the model has that property.
(the controller in blog will just be removed then)

You just need to properly implement something for the `.with_context(mail_create_nosubscribe=True)` part but maybe you can do that in JS instead in the small overide you will have here (to allow the duplicate capability, maybe also a getCopyContext method)